✨ feat: 스터디 공고 북마크 기능 구현

Jiwoong0322 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #229 
- 작업 요약: 사용자가 관심 있는 스터디 구인 공고를 저장(북마크)하고, 마이페이지 등 별도의 페이지에서 북마크한 공고 목록을 모아볼 수 있는 기능입니다. 사용자는 공고 목록 또는 상세 페이지에서 북마크를 추가하거나 취소할 수 있습니다.

## 📄 상세 내용
- [ ] 북마크 추가/삭제 및 목록 조회 API 구현 (REQ-RECM-010, REQ-RECM-011)
- [ ] 아키텍처 및 코드 품질 개선
- [ ] 테스트 및 안정성 확보

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Jiwoong0322 committed:
✨ feat: 스터디 공고 북마크 기능 구현

- 북마크 추가/삭제 및 목록 조회 API 구현 (REQ-RECM-010, REQ-RECM-011)
- 아키텍처 및 코드 품질 개선
- 테스트 및 안정성 확보

관련 이슈: #229

Jiwoong0322 committed:
♻️ refactor: isort, black 등 코드 리팩토링 진행

 - isort, black 진행
 - poetry install --no-root진행
 - mypy 테스트 성공

관련 이슈: #229

Meoyoug commented on apps/recruitments/views/bookmark_views.py:
공고는 외부 노출용으로 uuid를 사용할텐데 id 값으로 패스 파라미터를 받고있네요?

Meoyoug commented on apps/recruitments/views/bookmark_views.py:
```diff
@@ -0,0 +1,52 @@
+        try:
+            _, created = service.add(user=cast(User, request.user), recruitment_id=recruitment_id)
+            if created:
+                return Response({"detail": "북마크가 추가되었습니다."}, status=status.HTTP_201_CREATED)
```
성공 응답을 굳이 나눴어야 했나 싶긴한데 나눌 거라면 extend_schema로 스웨거에서 예상 응답확인할 수 있도록 해주세요

Meoyoug commented on apps/recruitments/views/bookmark_views.py:
```diff
@@ -0,0 +1,52 @@
+            service.remove(user=cast(User, request.user), recruitment_id=recruitment_id)
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        except NotFound as e:
+            return Response({"detail": str(e)}, status=status.HTTP_404_NOT_FOUND)
```
예외 메시지 처리 상세하게 써주세요

Jiwoong0322 committed:
♻️ refactor: uuid수정 및 응답 메시지 처리 리팩토링

 - recruitmentsId를 받던것을 uuid를 받도록 수정
 - 스웨거에서 예상 응답 받을 수 있도록 extend_schema추가
 예외 메시지를 error:에러메시지 로 통일하고 service에서 에러 진단, view에서 메시지를 전달하도록 함.

관련 이슈: #229

Jiwoong0322 committed:
♻️ refactor: id -> uuid로 바꾼 뒤 테스트 코드에 미적용한 것 리팩토링

 - 테스트 코드에서 URL생성시 id가 아닌 uuid로 생성하도록 수정.

관련 이슈: #229
